### PR TITLE
Report unsupported extended subgroup tests as skipped rather than passed

### PR DESCRIPTION
**Pull request by kpet**
Also don't check the presence of extensions for each sub-test.

Signed-off-by: Kévin Petit <kpet@free.fr>

### CODE REVIEW

File: test_conformance/subgroups/test_subgroup_ballot.cpp [hunk 1]
```diff
@@ -926,11 +926,15 @@ template <typename T> int run_non_uniform_broadcast_for_type(RunTestForType rft)
 int test_subgroup_functions_ballot(cl_device_id device, cl_context context,
                                    cl_command_queue queue, int num_elements)
 {
-    std::vector<std::string> required_extensions = { "cl_khr_subgroup_ballot" };
+    if (!is_extension_available(device, "cl_khr_subgroup_extended_types"))
```
Comment (gwawiork) [inline test_conformance/subgroups/test_subgroup_ballot.cpp:929]: Did you mean "cl_khr_subgroup_ballot" ?

File: test_conformance/subgroups/test_subgroup_ballot.cpp [hunk 2]
```diff
@@ -926,11 +926,15 @@ template <typename T> int run_non_uniform_broadcast_for_type(RunTestForType rft)
 int test_subgroup_functions_ballot(cl_device_id device, cl_context context,
                                    cl_command_queue queue, int num_elements)
 {
-    std::vector<std::string> required_extensions = { "cl_khr_subgroup_ballot" };
```
Comment (gwawiork) [inline test_conformance/subgroups/test_subgroup_ballot.cpp:929]: This variable is used in subhelpers.h removing makes dead code in that file.

**Review by gwawiork — CHANGES_REQUESTED** [review]
Have you considered make the only change in subherlpers.h file by setting TEST_SKIPPED_ITSELF ?

File: test_conformance/subgroups/test_subgroup_ballot.cpp (hunk 1 above)
Comment (kpet) [reply test_conformance/subgroups/test_subgroup_ballot.cpp]: Good catch, now fixed.

File: test_conformance/subgroups/test_subgroup_ballot.cpp (hunk 2 above)
Comment (kpet) [reply test_conformance/subgroups/test_subgroup_ballot.cpp:929]: Yes, might as well remove it all. Done.

**Comment by kpet** [comment]
Thanks for the review!

>Have you considered make the only change in subherlpers.h file by setting TEST_SKIPPED_ITSELF ?

Yes, that was my first change but it doesn't help with reducing the number of times the extension's presence is tested and reported.

File: test_conformance/subgroups/subhelpers.h [hunk 3]
```diff
@@ -1297,19 +1295,6 @@ template <typename Ty, typename Fns, size_t TSIZE = 0> struct test
...
-        for (std::string extension : test_params.required_extensions)
-        {
-            if (!is_extension_available(device, extension.c_str()))
-            {
-                log_info("The extension %s not supported on this device. SKIP "
-                         "testing - kernel %s data type %s\n",
-                         extension.c_str(), kname, TypeManager<Ty>::name());
-                return TEST_PASS;
-            }
-            kernel_sstr << "#pragma OPENCL EXTENSION " + extension
```
Comment (gwawiork) [inline test_conformance/subgroups/subhelpers.h:1309]: What about pragmas. Do we need use them or not ?

Comment (kpet) [reply test_conformance/subgroups/subhelpers.h:1309]: AFAICT the pragmas are not required by the extended subgroup specification. @bashbaug Can you confirm?

Comment (bashbaug) [reply test_conformance/subgroups/subhelpers.h:1309]: Correct, no pragmas are (currently) required for the extended subgroups extensions.

**Review by StuartDBrady — APPROVED** [review]
Looks good to me.

File: test_conformance/subgroups/subhelpers.h (hunk 3 above)
Comment (kpet) [reply test_conformance/subgroups/subhelpers.h:1309]: Thanks for confirming Ben!

**Pull request merged**